crates/extensions/c8y_mapper_ext/src/config_operations.rs

```diff
@@ -425,13 +425,11 @@ impl CumulocityConverter {
                         self.ops_dir.join(target.external_id.as_ref())
                     }
                     _ => {
-                        warn!("config_snapshot and config_update features for nested child devices are currently unsupported");
```

Comment:
They've been moved to that common logic which is validating all the operations at one place. The `wrap_errors` function that picks up the returned error will be logging it as an `error!("Mapping error")`. So, this would have been redundant. And when this unsupported error path is hit, it wouldn't have reached here anyway, as the new validation  function would return the error, before reaching here.

Reply:
I guessed it's no longer hittable, but then why do we still need to keep the `match` arms?

```rust
match &target.parent {
        Some(parent) if parent.is_default_main_device() => {
            // Support only first level child devices due to the limitation of our file system supported operations scheme.
            self.ops_dir.join(target.external_id.as_ref())
        }
        _ => {
...
```

Reply:
To build the op directory path in the normal case for immediate child devices and main device. For the nested child devices and service, it's a no-op, but those 2 specific paths won't be hit. Since match statements need to be exhaustive, I'm forced to keep those match arms with no-ops, even though they'll never be hit.